cpp/src/arrow/python/python-test.cc

```diff
@@ -167,5 +167,14 @@ TEST(BuiltinConversionTest, TestMixedTypeFails) {
   ASSERT_RAISES(UnknownError, ConvertPySequence(list, pool, &arr));
 }
 
+TEST_F(DecimalTest, FromPythonDecimalRescale) {
+  Decimal128 value;
+  OwnedRef python_decimal(this->CreatePythonDecimal("1.0134"));
+  auto type = ::arrow::decimal(10, 2);
+  const auto& decimal_type = static_cast<const DecimalType&>(*type);
+  ASSERT_RAISES(Invalid, internal::DecimalFromPythonDecimal(python_decimal.obj(),
```

Comment:
I guess this would truncate? Perhaps add a comment? :)

Reply:
No, this would fail. I'm asserting that it returns an `Invalid` status code.

Reply:
Yes, I meant it would fail because it refuses to truncate. I'm just suggesting to add a comment to make it clearer to the reader.

Reply:
I see :) Sure